webapp/src/http_handlers/sessions/sessions_tracker.rs

```diff
@@ -61,45 +87,108 @@ impl SessionsTracker
         }
     }
 
-    /// Builder method, that sets sessions life duration
+    /// ## Builder method, that sets sessions life duration.
+    /// ----
+    /// Args:
+    /// ---
+    /// * `duration` - new `life_duration` value.
     pub fn life_duration(mut self, duration: &Duration) -> Self
     {
         self.life_duration = duration.to_owned();
         self
     }
 
-    /// Returns a ref to a tracked session by its id
-    pub fn get_session(&self, id: &u128) -> Option<&Session>
+    /// ## Locks `sessions_data`.
+    /// ----
+    /// ## Returns:
+    /// `MutexGuard` for `sessions`. If the mutex is poisoned - it will end the process with the exit code of 1.
+    fn lock(&self) -> MutexGuard<HashMap<u128, Session>>
     {
-        self.sessions.get(&id)
+        self.sessions.lock().unwrap_or_else(|_| std::process::exit(1))
+    }
+    
+    /// ## Sets session's `source_path` field.
+    /// ----
+    /// Args:
+    /// ---
+    /// * `session_id` - id of the session to modify;
+    /// * `sourse_path` - new field value.
+    pub fn set_source_file(&self, session_id: &u128, source_path: &Path)
+    {
+        if let Some(session) = self.lock().get_mut(session_id)
+        {
+            session.set_source(source_path);
+        }
     }
 
-    /// Returns a mut ref to a tracked session by its id
-    pub fn get_mut_session(&mut self, id: &u128) -> Option<&mut Session>
+    /// ## Sets session's `last_connection` field.
+    /// ----
+    /// Args:
+    /// ---
+    /// * `session_id` - id of the session to modify;
+    /// * `new_date` - new field value.
+    pub fn set_last_connection(&self, session_id: &u128, new_date: DateTime<Utc>)
     {
-        self.sessions.get_mut(id)
+        let sessions = &mut self.lock();
+        if let Some(session) = sessions.get_mut(&session_id)
+        {
+            session.last_connection = new_date;
```

Comment:
`last_connection` seems to sound too general for this purpose, how about `last_connection_timestamp` or `timestamp_of_last_connection`?